dvc/repo/experiments/executor/local.py

```diff
@@ -213,3 +215,57 @@ def cleanup(self, infofile: str):
             checkpoint = self.scm.get_ref(EXEC_CHECKPOINT)
             if checkpoint and checkpoint != self._orig_checkpoint:
                 self.scm.set_ref(EXEC_APPLY, checkpoint)
+
+    @classmethod
+    def save(
+        cls,
+        info: "ExecutorInfo",
+        is_checkpoint: bool = False,
+        force: bool = False,
+    ) -> ExecutorResult:
+        from dvc.repo import Repo
+
+        exp_hash: Optional[str] = None
+        exp_ref: Optional[ExpRefInfo] = None
+
+        dvc = Repo(os.path.join(info.root_dir, info.dvc_dir))
+        old_cwd = os.getcwd()
+        if info.wdir:
+            os.chdir(os.path.join(dvc.scm.root_dir, info.wdir))
+        else:
+            os.chdir(dvc.root_dir)
+
+        try:
+            stages = dvc.commit([], force=force)
+            exp_hash = cls.hash_exp(stages)
+            cls.commit(
+                dvc.scm,
+                exp_hash,
+                exp_name=info.name,
+                checkpoint=is_checkpoint,
+            )
+            ref: Optional[str] = dvc.scm.get_ref(EXEC_BRANCH, follow=False)
+            exp_ref = ExpRefInfo.from_ref(ref) if ref else None
+            # TODO: research into how untracked files should be handled
+            if cls.WARN_UNTRACKED:
```

Comment:
> Add a flag to save that forces untracked files to be added included in the experiment

I am not sure what makes more sense from product perspective but I would expect untracked files to be included by default? c @dberenbaum

Reply:
We exclude untracked files from regular experiments, I would not expect `exp save` to include untracked files.

(including untracked files generally leads to unwanted behavior like git adding things like `__pycache__` directories that users forgot to gitignore)

Reply:
> We exclude untracked files from regular experiments, I would not expect exp save to include untracked files.

I was thinking that saving untracked in `exp save` could make sense because the use case is different. Some requirements in the original issue also sounded like referring to untracked files, i.e.:

> Code has already been run. The user might have already run code and wants to record it without committing it to their working branch.

Reply:
Whether exp saves or doesn't save untracked files, I think it should be consistent between `exp run` and `exp save`, and we need to make it very clear to users what the behavior is. Related: https://github.com/iterative/dvc.org/issues/4044.

Reply:
In order to use `exp.save` from a Python API (DVCLive scenario), it would be great if untracked files could be included (or an option exposed to force it). 

Otherwise, the first time DVCLive is run (or something new is logged that was not tracked before) the logged data will not be present in the exp ref:

```
from dvclive import Live

live = Live()
live.log_metric("foo", 1)

# This should call something like `dvc.api.make_experiment()`
# and the untracked files logged by DVCLive should be included
live.end()
```

Reply:
I think we can split between:
1. DVCLive-generated files: can DVCLive handle tracking these with Git?
2. Other untracked files: it's possible model artifacts or other files are being saved outside DVCLive, so it would be useful to at least have a flag to include these. I'd be wary of including everything in the repo by default.

Let's make 2 a follow up and worry about 1 for now.

Reply:
> DVCLive-generated files: can DVCLive handle tracking these with Git?

I guess so. It just feels odd.

> Let's make 2 a follow up and worry about 1 for now.

Ok, will implement 1 on DVCLive side